### PR TITLE
Add request() and http_client() helpers 

### PR DESCRIPTION
**Pull request by pyrech**

### CODE REVIEW

File: doc/06-helper.md [hunk 1]
```diff
@@ -112,6 +112,27 @@ You can check
 the [Symfony documentation](https://symfony.com/doc/current/components/finder.html)
 for more information about this class and how to use it.
 
+## HTTP Client
+
+The `http()` function returns an instance of
```
Comment (lyrixx) [inline doc/06-helper.md:117]: just wondering, `http()` or `http_client()` ?

File: src/Console/Application.php [hunk 2]
```diff
@@ -155,19 +156,12 @@ private function displayUpdateWarningIfNeeded(SymfonyStyle $symfonyStyle): void
...
-                'http' => [
-                    'method' => 'GET',
-                    'header' => 'User-Agent: castor',
-                    'timeout' => 5,
-                ],
-            ];
 
-            $context = stream_context_create($opts);
+            $response = http()->request('GET', 'https://api.github.com/repos/jolicode/castor/releases/latest', [
+                'timeout' => 5,
```
Comment (lyrixx) [inline src/Console/Application.php:161]:
```suggestion
                'timeout' => 1,
```

File: src/Console/Application.php [hunk 3]
```diff
@@ -155,19 +156,12 @@ private function displayUpdateWarningIfNeeded(SymfonyStyle $symfonyStyle): void
...
+            ]);
 
-            $content = @file_get_contents('https://api.github.com/repos/jolicode/castor/releases/latest', false, $context);
-
-            return json_decode($content ?: '', true) ?? [];
+            return $response->toArray();
```
Comment (lyrixx) [inline src/Console/Application.php:164]: Missing try / cacth { return null}

**Review by lyrixx — APPROVED** [review]
Top 👍🏼 

I just let a simple question

File: doc/06-helper.md (hunk 1 above)
Comment (joelwurtz) [reply doc/06-helper.md:117]: http() is fine for me, client is useless: it's implicit here

File: src/functions.php [hunk 4]
```diff
@@ -465,6 +466,11 @@ function finder(): Finder
     return new Finder();
 }
 
+function http(): HttpClientInterface
```
Comment (joelwurtz) [inline src/functions.php:469]: maybe we can allow to pass default options here for the http client ?

**Comment by pyrech** [comment]
PR updated

**Review by joelwurtz — APPROVED** [review]

File: src/Console/Application.php [hunk 5]
```diff
@@ -161,19 +162,16 @@ private function displayUpdateWarningIfNeeded(SymfonyStyle $symfonyStyle): void
...
+            $response = request('GET', 'https://api.github.com/repos/jolicode/castor/releases/latest', [
+                'timeout' => 1,
+            ]);
 
-            $content = @file_get_contents('https://api.github.com/repos/jolicode/castor/releases/latest', false, $context);
-
-            return json_decode($content ?: '', true) ?? [];
+            try {
+                return $response->toArray();
+            } catch (\Throwable) {
```
Comment (lyrixx) [inline src/Console/Application.php:172]: Can you narrow the scope of the exception?

**Review by lyrixx — APPROVED** [review]

**Pull request merged**